sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -656,6 +660,64 @@ trait ParquetPartitionReaderBase extends Logging with Arm with ScanWithMetrics
     }
   }
 
+  /**
+   * Convert cudf unsigned integer to wider signed integer that parquet expects
+   * After spark 3.2.0, parquet read uint8 as int16, uint16 as int32, uint32 as int64
+   *
+   * @param inputTable the input table
+   * @return converted table if the input table has unsigned integer
+   */
+  private def convertToUnsigned(inputTable: Table) = {
+    val typeCastingNeeded = needUnsignedCast(inputTable, readDataSchema)
+    if (typeCastingNeeded) {
+      val newColumns = new Array[ColumnVector](readDataSchema.length)
+      try {
+        withResource(inputTable) { table =>
+          (0 until readDataSchema.length).foreach(i => {
+            val readField = readDataSchema(i)
+            val origCol = table.getColumn(i)
+            val col: ColumnVector =
+              ColumnCastUtil.ifTrueThenDeepConvertTypeAtoTypeB(origCol, readField.dataType,
+                (dt, cv) => needUnsignedToSignedCastFunction(cv, dt),
+                (dt, cv) => cv.castTo(
+                  DType.create(GpuColumnVector.getNonNestedRapidsType(dt).getTypeId)))
+            newColumns(i) = col
+          })
+        }
+        new Table(newColumns: _*)
+      } finally {
+        newColumns.safeClose()
+      }
+    } else {
+      inputTable
+    }
+  }
+
+  def needUnsignedCast(table: Table, schema: StructType): Boolean = {
+    (0 until table.getNumberOfColumns).exists(i =>
+      needUnsignedToSignedCast(table.getColumn(i), schema.fields(i).dataType))
+  }
+
+  def needUnsignedToSignedCast(cv: ColumnView, dataType: DataType): Boolean = {
+    dataType match {
+      case a: ArrayType =>
+        val child = cv.getChildColumnView(0)
```

Comment:
This "child" is leaked, needing to be closed.

Reply:
No needed now, moved to the original columns iterating logic